src/parsing/processProlog.ts

```diff
@@ -88,6 +88,12 @@ export default function processProlog(
 
 		staticContext.registerNamespace(prefix, namespaceURI);
 	});
+	astHelper.getChildren(prolog, 'defaultNamespaceDecl').forEach(defaultNamespaceDecl => {
+		const namespaceURI = astHelper.getTextContent(
+			astHelper.getFirstChild(defaultNamespaceDecl, 'uri')
+		);
+		staticContext.setDefaultFunctionNamespace(namespaceURI);
```

Comment:
Can we add a check for the disallowed namespaces first? You could pretty much copy that from the handling of the `namespaceDecl`.
[Spec](https://www.w3.org/TR/xquery-31/#doc-xquery31-DefaultNamespaceDecl) says those same namespaces should also raise XQST0070.